lxd/resource_lxd_instance.go

```diff
@@ -0,0 +1,896 @@
+				Default:  "",
+			},
+
+			"target": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Optional: true,
+				Computed: true,
+			},
+
+			"image": {
+				Type:             schema.TypeString,
+				ForceNew:         true,
+				Required:         true,
+				DiffSuppressFunc: suppressImageDifferences,
+			},
+
+			"profiles": {
+				Type:     schema.TypeList,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Optional: true,
+				ForceNew: false,
+				Computed: true,
+			},
+
+			"device": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						"type": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: resourceLxdValidateDeviceType,
+						},
+
+						"properties": {
+							Type:     schema.TypeMap,
+							Required: true,
+						},
+					},
+				},
+			},
+
+			"config": {
+				Type:     schema.TypeMap,
+				Optional: true,
+			},
+
+			"limits": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				ForceNew: false,
+			},
+
+			"ephemeral": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+				ForceNew: true,
+			},
+
+			"wait_for_network": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+				ForceNew: false,
+			},
+
+			"privileged": {
+				Type:       schema.TypeBool,
+				Optional:   true,
+				Default:    false,
+				ForceNew:   false,
+				Deprecated: "Use a config setting of security.privileged=1 instead",
+			},
+
+			"file": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"content": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+
+						"source": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+
+						"target_file": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						"uid": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+
+						"gid": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+
+						"mode": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  "0755",
+						},
+
+						"create_directories": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+					},
+				},
+			},
+
+			"ip_address": {
+				Type:     schema.TypeString,
+				Computed: true,
+				ForceNew: false,
+			},
+
+			"ipv4_address": {
+				Type:     schema.TypeString,
+				Computed: true,
+				ForceNew: false,
+			},
+
+			"ipv6_address": {
+				Type:     schema.TypeString,
+				Computed: true,
+				ForceNew: false,
+			},
+
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"mac_address": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"start_instance": {
```

Comment:
Since we're renaming this anyway, please rename this to `start_on_create` to clarify that it is only use to control the creation state. Also, if you don't mind, add it to the documentation. If not, I can add it after merge.

Reply:
Done.